### PR TITLE
[Merged by Bors] - feat: port RingTheory.Polynomial.Quotient

### PR DESCRIPTION
**Pull request by jjaassoonn**
Coming from #3292

- [x] depends on: #3414

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3414
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3414~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/RingTheory/Polynomial/Quotient.lean [hunk 1]
```diff
@@ -54,183 +64,228 @@ open Polynomial
...
   refine' mul_eq_zero_of_left (Polynomial.ext fun m => _) (X ^ n)
   erw [coeff_C]
   by_cases h : m = 0
-  · simpa [h] using quotient.eq_zero_iff_mem.2 ((mem_map_C_iff.1 ha) n)
+  · simpa [h] using Quotient.eq_zero_iff_mem.2 ((mem_map_C_iff.1 ha) n)
   · simp [h]
-#align ideal.eval₂_C_mk_eq_zero Ideal.eval₂_c_mk_eq_zero
+#align ideal.eval₂_C_mk_eq_zero Ideal.eval₂_C_mk_eq_zero
+
+lemma polynomialQuotientEquivQuotientPolynomial_eval₂RingHom_leftInverse (I : Ideal R) :
```
Comment (jcommelin) [inline Mathlib/RingTheory/Polynomial/Quotient.lean:90]: Could you please leave a `-- Porting note: ` explaining why you pulled out these proofs into standalone lemmas?

Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/Polynomial/Quotient.lean]: I restored this one, but `quotientEquivQuotientMvPolynomial` is still too slow; added a porting note for that.

File: Mathlib/RingTheory/Polynomial/Quotient.lean [hunk 2]
```diff
@@ -105,7 +104,7 @@ def polynomialQuotientEquivQuotientPolynomial (I : Ideal R) :
   map_add' f g := by simp only [eval₂_add, coe_eval₂RingHom]
   left_inv := by
     intro f
-    refine Polynomial.induction_on' f ?_ ?_
+    refine' Polynomial.induction_on' f _ _
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Polynomial/Quotient.lean:107]: This change goes in the wrong direction, so I'll revert it.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**